app/controllers/users_controller.rb

```diff
@@ -19,6 +19,25 @@ def edit
     @user = UsersFacade.find_user(params[:id])
   end 
 
+  def update
+    if params[:password] == params[:password_confirmation]
+      user = UsersFacade.find_user(params[:id].to_i)
+      response = UsersFacade.update_user(user, params[:name], params[:email], params[:password])
+      session[:user_id] = response[:user_id].to_i
+      if response[:status] == 202
+        flash[:success] = "Profile Updated"
+        redirect_to user_path(current_user)
+      else
+        flash[:error] = response[:error].join(', ')
+        redirect_back(fallback_location: edit_user_path(current_user))
+      end
+    else
+      session[:user_id] = params[:id]
+      flash[:error] = "Passwords Must Match"
+      redirect_back(fallback_location: edit_user_path(current_user))
+    end
+  end
+
```

Comment:
This looks great and the sessions will be useful for when we start building Oauth. We may need to add something for password confirmation on the BE and this is something we can look into.